code/game/objects/structures/statues.dm

```diff
@@ -342,3 +397,181 @@
 	..()
 	qdel(src)
 
+
+////////////////////////////////
+
+/obj/item/chisel
+	name = "chisel"
+	desc = "A carving tool."
+	icon = 'icons/obj/statue.dmi'
+	icon_state = "chisel"
+	flags = CONDUCT
+	slot_flags = SLOT_BELT
+	force = 5
+	w_class = WEIGHT_CLASS_TINY
+	throwforce = 5
+	throw_speed = 3
+	throw_range = 5
+	materials = list(MAT_METAL=75)
+	attack_verb = list("stabbed")
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	usesound = 'sound/items/gavel.ogg'
+	var/tool_speed = 100
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 30)
+
+
+/obj/item/chisel/afterattack(atom/target, mob/living/carbon/user)
```

Comment:
Currently, with the way this is set up, when you use a chisel on a block, you hit the block with a *clunk* sound. While its not the end of the world, I think this should be set up a bit differently. Instead of handling all this in the chisel afterattack code, handle it in carving `attackby`:

```js
/obj/structure/carving/attackby(obj/item/W, mob/living/user, params)
    ...
    if(istype(W, /obj/item/chisel))
	get_chiseled(user, W)
	return TRUE // This will prevent you from whacking the block with the chisel.
```
Then in `get_chiseled` proc (You can use whatever proc name you want):
```js
/obj/structure/carving/proc/get_chiseled(mob/user, obj/item/chisel/C)
    . = ..()
    var/list/availablestatues = typesof(text2path("/obj/structure/carving/statue/[material]"))
    availablestatues -= text2path("/obj/structure/carving/statue/[B.material]")
    ...
    // Put the rest of the previous chisel afterattack code here.
```
Obviously the code in `get_chiseled` needs to be changed around a bit because the `src` object is a `/carving` and not `/chisel` but yeah, hopefully you get the idea.

Reply:
Yeah I thought about those previous possibilities but then I'd need to add additionnal checks to the attackby code to make sure it's actually a block being attacked and not a different subtype, etc etc.

Reply:
What if you just override the chisel proc for the subtypes you don't want to be chisel-able like so:
```js
/obj/structure/carving/block/get_chiseled(..)
    return

/obj/structure/carving/statue/get_chiseled(..)
    return
```